templates/eks-protection-stack.yml

```diff
@@ -0,0 +1,536 @@
+          - Name: IMAGE_REGION
+            Type: PLAINTEXT
+            Value: !If [ UseECR, !Ref AWS::Region, 'CrowdStrike' ]
+          - Name: NODE_SENSOR_REPO
+            Type: PLAINTEXT
+            Value: !If [ UseECR, !Ref NodeSensorRepo, 'CrowdStrike' ]
+          - Name: NODE_SENSOR_URI
+            Type: PLAINTEXT
+            Value: !If [ UseECR, !GetAtt NodeSensorRepo.RepositoryUri, 'CrowdStrike' ]
+          - Name: ACCOUNT_ID
+            Type: PLAINTEXT
+            Value: lambda
+          - Name: REGION
+            Type: PLAINTEXT
+            Value: lambda
+          - Name: SWITCH_ROLE
+            Type: PLAINTEXT
+            Value: !Ref EKSExecutionRoleName
+          - Name: NAT_IP
+            Type: PLAINTEXT
+            Value: !GetAtt NatGatewayEIP.PublicIp
+        Image: aws/codebuild/standard:7.0
+        PrivilegedMode: true
+        Type: LINUX_CONTAINER
+      VpcConfig:
+        SecurityGroupIds: 
+          - !GetAtt NoIngressSecurityGroup.GroupId
+        Subnets: 
+          - !GetAtt PrivateSubnet1.SubnetId
+        VpcId: !GetAtt PrivateSubnet1.VpcId
+      Artifacts:
+        Type: NO_ARTIFACTS
+      Name: !Ref CodeBuildProjectName
+      ServiceRole: !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${CodeBuildRoleName}
+      Source: 
+        Location: !Sub '${StagingS3Bucket}/${SourceS3BucketNamePrefix}/codebuild/codebuild.zip'
+        Type: S3
+      TimeoutInMinutes: 90
+      Visibility: PRIVATE
+
+# Secret
+  SMFalconAPIKey:
+    Type: AWS::SecretsManager::Secret
+    Metadata:
+      checkov:
+        skip:
+          - id: CKV_AWS_149
+            comment: The default key aws/secretsmanager is sufficient to secure this resource
+    Properties:
+      Name: crowdstrike-eks-secret
+      Description: 'Client ID and Secret for the Falcon API'
+      SecretString: !Sub |
+        {
+          "client_id": "${FalconClientId}",
+          "client_secret": "${FalconClientSecret}",
+          "cid": "${FalconCID}",
+          "cs_cloud":"${CrowdStrikeCloud}",
+          "docker_api_token":"${DockerAPIToken}"
+        }
+
+ # Create ECR Repositories for Falcon Images
+  NodeSensorRepo:
+    Type: AWS::ECR::Repository
+    Condition: UseECR
+    Properties:
+      EmptyOnDelete: true
+      RepositoryName: crowdstrike/falcon-sensor
+      RepositoryPolicyText:
+        Version: "2012-10-17"
+        Statement: 
+          - 
+            Sid: AllowPushPull
+            Effect: Allow
+            Principal: 
+              AWS: 
+                - !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${CodeBuildRoleName}
+            Action: 
+              - "ecr:GetDownloadUrlForLayer"
+              - "ecr:BatchGetImage"
+              - "ecr:BatchCheckLayerAvailability"
+              - "ecr:PutImage"
+              - "ecr:InitiateLayerUpload"
+              - "ecr:UploadLayerPart"
+              - "ecr:CompleteLayerUpload"
+  KPAgentRepo:
+    Type: AWS::ECR::Repository
```

Comment:
**Recommendation:** _ECR Repository should have ScanOnPush enabled_
Add exemption with valid reason for not enabling this option.